kite-tutorials/crunch/src/org/kitesdk/examples/tutorials/crunch/AggregateEvents.java

```diff
@@ -0,0 +1,152 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kitesdk.examples.tutorials.crunch;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.util.Calendar;
+import java.util.Iterator;
+import java.util.TimeZone;
+import org.apache.crunch.DoFn;
+import org.apache.crunch.Emitter;
+import org.apache.crunch.MapFn;
+import org.apache.crunch.PCollection;
+import org.apache.crunch.Pair;
+import org.apache.crunch.Target;
+import org.apache.crunch.types.avro.Avros;
+import org.apache.crunch.util.CrunchTool;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.ToolRunner;
+import org.kitesdk.data.Dataset;
+import org.kitesdk.data.Datasets;
+import org.kitesdk.data.crunch.CrunchDatasets;
+import org.kitesdk.data.event.StandardEvent;
+import org.kitesdk.data.spi.filesystem.FileSystemDatasets;
+import org.kitesdk.examples.demo.event.Session;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AggregateEvents extends CrunchTool implements Serializable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AggregateEvents.class);
+
+  @Override
+  public int run(String[] args) throws Exception {
+	  
+    // Turn debug on while in development.
+    getPipeline().enableDebug();
+    getPipeline().getConfiguration().set("crunch.log.job.progress", "true");
+    
+    // Step 1. Load the dataset into the Pipeline.
+    Dataset<StandardEvent> eventsDataset = Datasets.load(
+        "dataset:hive:events", StandardEvent.class);
+
+	// If the dataset is empty, stop.
+    if (eventsDataset.isEmpty()) {
+      LOG.info("No records to process.");
+      return 0;
+    }
+
+    // Create a parallel collection from the Kite CrunchDatasets
+    // Source "events" dataset.
+    PCollection<StandardEvent> events = read(
+        CrunchDatasets.asSource(eventsDataset));
+
+    /* Step 2. Process the data.
+     * a. Create a session key by combining the user ID and session ID.
+     * b. Group together all events with the same session key.
+     * c. For each group, create a Session record as an Avro Specific object. 
+     */
+    PCollection<Session> sessions = events
+        .by(new GetSessionKey(), Avros.strings())
+        .groupByKey()
+        .parallelDo(new MakeSession(), Avros.specifics(Session.class));
+
+    // Step 3. Append the derived sessions to the Kite CrunchDatasets
+    // Target "sessions" dataset.
+    getPipeline().write(sessions,
+        CrunchDatasets.asTarget("dataset:hive:sessions"),
```

Comment:
This creates a Crunch Target that is configured to write to the sessions dataset in Hive. The rest of the statement instructs Crunch to append the (parallel) collection of sessions to that target. The way you phrase Step 3 here doesn't quite explain what is happening and why the word "target" is introduced. I'd like to make it a little more clear that a Target is Crunch's notion of a place to put data and the CrunchDatasets utility method is returning the given dataset (by URI) as a Target for Crunch.